src/main/osd/osd_elements.c

```diff
@@ -1127,6 +1127,55 @@ static void osdElementEfficiency(osdElementParms_t *element)
         tfp_sprintf(element->buff, "----%c/%c", SYM_MAH, unitSymbol);
     }
 }
+
+int16_t osdGetBoundaryFromDisplayX(int16_t error_x)
+{
+    if (error_x < -(osdConfig()->camera_fov_h / 2)){
+        error_x = -(osdConfig()->camera_fov_h / 2);
+    } else if (error_x >= osdConfig()->camera_fov_h / 2){
+        error_x = osdConfig()->camera_fov_h / 2;
+    }
+
+    return error_x;
+}
+
+int16_t osdGetBoundaryFromDisplayY(int16_t error_y)
+{
+    if (error_y < -(osdConfig()->camera_fov_v / 2)){
+        error_y = -(osdConfig()->camera_fov_v / 2);
+    } else if (error_y >= osdConfig()->camera_fov_v / 2){
+        error_y = osdConfig()->camera_fov_v / 2;
+    }
+
+    return error_y;
+}
+
+static void osdElementMovingHomePoint(osdElementParms_t *element)
```

Comment:
```suggestion
static void osdElementProjectHomePoint(osdDisplayPost_t *port, osdElementParms_t *element)
```

Reply:
Perfekt, will have a look thx

Reply:
You can use the displayport pointer in the function like `port->cols` for example

Reply:
I cant get it working:
```
static void osdElementProjectHomePoint(displayPort_t *osdDisplayPort, osdElementParms_t *element)
{
    if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
        int16_t center_x = osdDisplayPort->cols / 2;
        int16_t center_y = osdDisplayPort->rows / 2;
.....
```

I get an error on compiling, is it because of 2 pointers?

```
./src/main/osd/osd_elements.c: In function 'osdElementProjectHomePoint':
./src/main/osd/osd_elements.c:1140:60: error: 'deviceType' undeclared (first use in this function)
 1140 |         displayPort_t *osdDisplayPort = osdGetDisplayPort(&deviceType);
      |                                                            ^~~~~~~~~~
./src/main/osd/osd_elements.c:1140:60: note: each undeclared identifier is reported only once for each function it appears in
```

Reply:
If I compile it I get an error and if I directly put it in the function with the pointer, the osd output is not anymore shown on the osd, so there is no output anymore. Have not enough experience 😢

Reply:
No worries, I'll have a look and tell you what to change

Reply:
Ok thx

Reply:
I will check it on Sunday thx.